mhs/outbound/outbound/request/synchronous/handler.py

```diff
@@ -132,30 +134,47 @@ def _should_invoke_sync_async_workflow(self, interaction_config, sync_async_head
         else:
             return False
 
-    async def _invoke_sync_async(self, message_id, correlation_id, interaction_details, body, async_workflow):
+    async def _invoke_sync_async(self, from_asid, message_id, correlation_id, interaction_details, body, async_workflow):
         sync_async_workflow: workflow.SyncAsyncWorkflow = self.workflows[workflow.SYNC_ASYNC]
-        status, response, wdo = await sync_async_workflow.handle_sync_async_outbound_message(message_id,
+        status, response, wdo = await sync_async_workflow.handle_sync_async_outbound_message(from_asid,
+                                                                                             message_id,
                                                                                              correlation_id,
                                                                                              interaction_details,
                                                                                              body,
                                                                                              async_workflow)
-        await self.return_sync_async_response(status, response, wdo)
+        await self.write_response_with_store_updates(status, response, wdo,
+                                                     wd.MessageStatus.OUTBOUND_SYNC_ASYNC_MESSAGE_SUCCESSFULLY_RESPONDED,
+                                                     wd.MessageStatus.OUTBOUND_SYNC_ASYNC_MESSAGE_FAILED_TO_RESPOND)
 
-    async def return_sync_async_response(self, status: int, response: str, wdo: wd.WorkDescription):
+    async def write_response_with_store_updates(self, status: int, response: str, wdo: wd.WorkDescription,
+                                                success, failure):
         try:
             self._write_response(status, response)
-            await wdo.set_outbound_status(wd.MessageStatus.OUTBOUND_SYNC_ASYNC_MESSAGE_SUCCESSFULLY_RESPONDED)
+            if wdo:
```

Comment:
You need some test cases to validate that this method handles not being passed a `wdo`.